cmake/thirdparty/get_libcudacxx.cmake

```diff
@@ -13,16 +13,22 @@
 # =============================================================================
 
 # Use CPM to find or clone libcudacxx
-function(find_and_configure_libcudacxx)
-    include(${rapids-cmake-dir}/cpm/libcudacxx.cmake)
-
-    set(exports BUILD_EXPORT_SET cuco-exports)
-    if(INSTALL_CUCO)
-      list(APPEND exports INSTALL_EXPORT_SET cuco-exports)
-    endif()
-
-    rapids_cpm_libcudacxx(${exports})
-    set(LIBCUDACXX_INCLUDE_DIR "${libcudacxx_SOURCE_DIR}/include" PARENT_SCOPE)
+function(find_and_configure_libcudacxx VERSION)
+  rapids_cpm_find(
+    libcudacxx         ${VERSION}
+    CPM_ARGS
+      GIT_REPOSITORY https://github.com/NVIDIA/libcudacxx.git
+      GIT_TAG        ${VERSION}
+      GIT_SHALLOW    TRUE
+      DOWNLOAD_ONLY  TRUE
+  )
+  # TODO: Once libcu++ exports a target, use that instead
+  add_library(libcudacxx INTERFACE)
+  target_include_directories(libcudacxx
+    INTERFACE   $<BUILD_INTERFACE:${libcudacxx_SOURCE_DIR}/include>
+                $<INSTALL_INTERFACE:include/cuco/libcudacxx>)
+  install(DIRECTORY ${libcudacxx_SOURCE_DIR}/include/ DESTINATION include/cuco/libcudacxx)
+  install(DIRECTORY ${libcudacxx_SOURCE_DIR}/libcxx/include/ DESTINATION include/cuco/libcxx/include)
 endfunction()
 
-find_and_configure_libcudacxx()
+find_and_configure_libcudacxx(1.9.1)
```

Comment:
The 1.9.1 tag is broken, you have to use branch-1.9.1.

Which is what rapids-cmake 23.02 is using.

Reply:
Specifically the atomic headers are broken since they reference types that don't exist, due to a incorrect back porting

Reply:
Hmmm. Looks like libcudacxx deleted and recreated the 1.9.1 tag so new clones of the repo now have the corrected atomic version but older clones won't.

Reply:
> Which is what rapids-cmake 23.02 is using.

Understood. The main goal of this PR is to make libcudacxx version in cuco not depend on the version defined by `rapids-cmake`.

Reply:
That will break downstream projects which also bring in libcudacxx since we will either get the wrong version or have ODR violations due to differing implementations of the same functions

Reply:
Oh no :exploding_head: 

Can we link downstream projects to a cuco-favorable libcudacxx version to solve the conflict?
cc @jrhemstad